Tests/ClientRuntimeTests/NetworkingTests/DataStreamReaderTests.swift

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+@testable import ClientRuntime
+
+final class DataStreamReaderTests: XCTestCase {
+    func testRead() throws {
+        let sut = DataStreamReader(data: Data([0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A]))
```

Comment:
Nit: Can we move this out of the test so that it can be reused?

Reply:
I moved the data part but not the reader as we don't one test interfere another.